PostHogTests/PostHogSDKTest.swift

```diff
@@ -847,6 +847,98 @@ class PostHogSDKTest: QuickSpec {
+
+            _ = sut.getFeatureFlag("some_key")
+
+            let event = getBatchedEvents(server)
+            expect(event.first!.event).to(equal("$feature_flag_called"))
+        }
+
+        it("captures a second $feature_flag_called if reloaded flag changed value ") {
+            let sut = self.getSut(
+                preloadFeatureFlags: false,
+                sendFeatureFlagEvent: true,
+                flushAt: 2
+            )
+
+            sut.reloadFeatureFlags {
+                DispatchQueue.main.async {
+                    // this should be captured with value true
+                    _ = sut.getFeatureFlag(server.flagKey)
+                }
+                server.flagValue = false
+                sut.reloadFeatureFlags {
+                    DispatchQueue.main.async {
+                        // this should be captured with value false
+                        _ = sut.getFeatureFlag(server.flagKey)
+                    }
+                }
+            }
+
+            let batches = getBatchedEvents(server)
+            expect(batches.count).to(equal(2))
+            expect(batches[0].event).to(equal("$feature_flag_called"))
+            expect(batches[1].event).to(equal("$feature_flag_called"))
+            expect(batches[0].properties["$feature_flag_response"] as? Bool).to(beTrue())
+            expect(batches[1].properties["$feature_flag_response"] as? Bool).to(beFalse())
+        }
+
+        it("does not capture a second $feature_flag_called if reloaded flag did not change value ") {
+            let sut = self.getSut(
+                preloadFeatureFlags: false,
+                sendFeatureFlagEvent: true,
+                flushAt: 2
+            )
+
+            sut.reloadFeatureFlags {
+                DispatchQueue.main.async {
+                    // this should be captured
+                    _ = sut.getFeatureFlag(server.flagKey)
+                }
+                sut.reloadFeatureFlags {
+                    DispatchQueue.main.async {
+                        // this should not be captured
+                        _ = sut.getFeatureFlag(server.flagKey)
+                        sut.flush()
+                    }
+                }
+            }
+
+            let batches = getBatchedEvents(server)
+            expect(batches.count).to(equal(1))
+            expect(batches[0].event).to(equal("$feature_flag_called"))
+            expect(batches[0].properties["$feature_flag_response"] as? Bool).to(beTrue())
+        }
+
+        it("does captures a second $feature_flag_called if reloaded flag did not change value, but keys were not preloaded") {
+            let sut = self.getSut(
+                preloadFeatureFlags: false,
+                sendFeatureFlagEvent: true,
+                flushAt: 2
+            )
+
+            // this should capture with value nil
+            _ = sut.getFeatureFlag(server.flagKey)
+
+            sut.reloadFeatureFlags {
+                DispatchQueue.main.async {
```

Comment:
why do we use dispatch queue here?

Reply:
the event is sent async thats why the test will fail, `getBatchedEvents(server)` will await so test will succeed, if you need to read flags before calling this and during `reloadFeatureFlags`, you will need another approach, I think using ` DispatchQueue.main.async` here will make the test flaky because its not guaranteed that executes before the test ends

Reply:
the call to `getBatchedEvents` is before you do `DispatchQueue.main.async` so in this case when you call `DispatchQueue.main.async` the async request was likely still happening, you'd need to call `getBatchedEvents` before, that's the method that awaits the async processes

Reply:
or even better, `let requests = getDecideRequest(server)` should await all pendings requests to the decide api